packages/extension-commerce-bm-seo/src/setup-app.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+    ApplicationExtension,
+    SliceInitializer,
+    withApplicationExtensionStore
+} from '@salesforce/pwa-kit-extension-sdk/react'
+import {applyHOCs} from '@salesforce/pwa-kit-extension-sdk/react/utils'
+import {
+    AsyncRouteProps,
+    ComponentMap,
+    GetRoutesParams,
+    BeforeRouteMatchParams
+} from '@salesforce/pwa-kit-extension-sdk/types'
+
+// Local Imports
+import {Config} from './types'
+import {
+    createPlaceholderComponent,
+    getComponentForUrlMapping,
+    withPropsWrapper
+} from './utils/component-utils'
+import {getAppOrigin, getShopperSeoClient} from './utils/shopper-seo-utils'
+
+// Others
+import extensionMeta from '../extension-meta.json'
+
+interface StoreSlice {
+    count: number
+    increment: () => void
+    decrement: () => void
+}
+
+// This is safe to delete if your extension does not use state. If you aren't using this, ensure you remove the
+// `withApplicationExtensionStore` usage below as well.
+const sliceInitializer: SliceInitializer<StoreSlice> = (set) => ({
+    count: 0,
+    increment: () => set((state) => ({count: state.count + 1})),
+    decrement: () => set((state) => ({count: state.count - 1}))
+})
+
+class CommerceBmSeo extends ApplicationExtension<Config> {
+    static readonly id = extensionMeta.id
+
+    /**
+     * Use this method to wrap or enhance your PWA-Kit application using [React higher-order components](https://legacy.reactjs.org/docs/higher-order-components.html).
+     * You can use this to add visual treatments to your application, change the props that are supplied to the application component
+     * or add things like providers and contexts to be used throughout your app.
+     */
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
+        const HOCs = [
+            // Optionally include state for this extension using `withApplicationExtensionStore`
+            (component: React.ComponentType<any>) =>
+                withApplicationExtensionStore(component, {
+                    id: extensionMeta.id,
+                    initializer: sliceInitializer
+                })
+        ]
+
+        return applyHOCs(App, HOCs)
+    }
+
+    /**
+     * This method is used to make changes to the PWA-Kit application routes. If your extension adds a new page to the application
+     * then you can add it to the router here. The routes passed to this method is an accrued list of routes that have been added
+     * from extensions applied before it. It is called during the `getRoutes` phase on both the server and the client.
+     *
+     * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
+     */
+    async getRoutesAsync({locals}: GetRoutesParams): Promise<AsyncRouteProps[]> {
+        let urlMapping
+
+        // TODO: do we need this check?
+        if (!locals.originalUrl) {
+            return Promise.resolve([])
+        }
```

Comment:
That is most likely guaranteed to be there. So I'm not sure if it's required.But if we want a check, I'm wondering if we want to throw an error instead.